### PR TITLE
refactor: remove direct msgraph dependency, updated dependencies

### PR DESCRIPTION
**Pull request by fmartingr**
#### Summary
- Updates mscalendar to v1.3.0
- Removes dead code
- Refactored package file names and code
- Removed `msgraph` as a direct dependency (as a result of the above)
- Upgraded rest of dependencies

### CODE REVIEW

File: gcal/gcal.go [hunk 1]
```diff
@@ -28,3 +36,64 @@ func GetGcalProviderConfig() config.ProviderConfig {
...
+}
+
+func (c *client) CallFormPost(method, url string, in url.Values, out interface{}) (responseData []byte, err error) {
+	contentType := "application/x-www-form-urlencoded"
+	buf := strings.NewReader(in.Encode())
+	return c.call(method, url, contentType, buf, out)
+}
+
+func (c *client) call(method, callURL, contentType string, inBody io.Reader, out interface{}) (responseData []byte, err error) {
+	req, err := http.NewRequest(method, callURL, inBody)
```
Comment (mickmister) [inline gcal/gcal.go:57]: Maybe this file can be called something like `http_request.go`? Some name to show that the main purpose is sending outgoing requests through the `net/http` library

**Review by mickmister — APPROVED** [review]

File: gcal/gcal.go (hunk 1 above)
Comment (fmartingr) [reply gcal/gcal.go]: 👍  https://github.com/mattermost/mattermost-plugin-google-calendar/pull/77/commits/8ceae32aeebafcd0e4a1e9e889d935343c0eb47a

**Review by mickmister — APPROVED** [review]

File: gcal/http_request.go
Comment (hanzei) [inline gcal/http_request.go:1]: 2/5 this methods could just be part of `gcal/client.go`.

**Review by hanzei — CHANGES_REQUESTED** [review]
NIce cleanup :+1: , one suggestion below

**Review by hanzei — APPROVED** [review]

File: gcal/http_request.go
Comment (fmartingr) [reply gcal/http_request.go:1]: That's actually better, yeah. Was not convinced about that http file.

**Pull request merged**